runtime/jvmti/jvmtiThread.c

```diff
@@ -1344,8 +1353,8 @@ jvmtiSuspendResumeCallBack(J9VMThread *vmThread, J9MM_IterateObjectDescriptor *o
 					}
 				} else {
 					/* targetThread is NULL only for virtual threads. */
-					if (0 != J9OBJECT_U32_LOAD(currentThread, vthread, vm->isSuspendedByJVMTIOffset)) {
-						J9OBJECT_U32_STORE(currentThread, vthread, vm->isSuspendedByJVMTIOffset, 0);
+					if (0 != J9OBJECT_U32_LOAD(currentThread, vthread, vm->isSuspendedInternalOffset)) {
+						J9OBJECT_U32_STORE(currentThread, vthread, vm->isSuspendedInternalOffset, 0);
 					}
```

Comment:
The halt flag should only be cleared if the vthread is mounted. `isSuspendedInternal` should always be set. There is also an error in existing code.
```
	if ((NULL != targetThread) && (vthread == targetThread->threadObject)) {
		...
	}
	J9OBJECT_U32_STORE(vmThread, vthread, vm->isSuspendedInternalOffset, 0);
```

Reply:
added code to clear the halt flag (J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND) only if the thread
is mounted, i.e., threadObject == targetThread->threadObject in J9VMThread's publicFlags